package.json

```diff
@@ -167,46 +168,34 @@
     "test:e2e": "yarn jest --config ./test/jest.config.e2e.js",
     "test:all": "yarn run test && yarn run test:functional && yarn run test:e2e & yarn run test:e2e:pkg",
     "pre:ci": "yarn run lint",
-    "coverage:publish": "codecov",
     "lint": "yarn run type-check && yarn run lint:ts",
-    "lint:ts": "eslint \"**/*.{js,jsx,ts,tsx}\"",
+    "lint:ts": "eslint \"**/*.{js,jsx,ts,tsx}\" -c ./eslintrc.js",
     "lint:lockfile": "lockfile-lint --path yarn.lock --type yarn --validate-https --allowed-hosts verdaccio npm yarn",
     "start": "yarn babel-node --extensions \".ts,.tsx\" src/lib/cli",
     "start:debug": "yarn node debug/bootstrap.js",
     "code:build": "yarn babel src/ --out-dir build/ --copy-files --extensions \".ts,.tsx\" --source-maps inline",
     "code:docker-build": "yarn babel src/ --out-dir build/ --copy-files --extensions \".ts,.tsx\"",
     "docker": "docker build -t verdaccio/verdaccio:pr-2122 . --no-cache",
-    "docker:run": "docker run -it --rm -p 4873:4873 verdaccio/verdaccio:local"
+    "docker:run": "docker run -it --rm -p 4873:4873 verdaccio/verdaccio:local",
+    "postinstall": "husky install",
+    "prepublishOnly": "pinst --disable",
+    "postpublish": "pinst --enable"
```

Comment:
Yarn 2 doesn't run these scripts (https://yarnpkg.com/advanced/lifecycle-scripts), you can add a small plugin to do this on publish

```js
module.exports = {
	name: `plugin-postinstall`,
	factory: () => ({
		hooks: {
			beforeWorkspacePacking(workspace, data) {
				if ('scripts' in data && 'postinstall' in data.scripts) {
					delete data.scripts.postinstall;
				}
			},
		},
	}),
};

```

Reply:
husky@6 for some reason does not skip the install withing the docker image. I've tried with several solutions from the docs, nothing. I can skip this since is kinda optional for me.

What bugs me now, is the Docker image 

```
docker run -it --rm --name verdaccio -p 4873:4873 verdaccio/verdaccio:update-yarn-pnp

internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'pino'
Require stack:
- /opt/verdaccio/build/lib/logger/logger.js
- /opt/verdaccio/build/lib/logger/index.js
- /opt/verdaccio/build/lib/cli.js
- /opt/verdaccio/bin/verdaccio
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/opt/verdaccio/build/lib/logger/logger.js:11:36)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/verdaccio/build/lib/logger/logger.js',
    '/opt/verdaccio/build/lib/logger/index.js',
    '/opt/verdaccio/build/lib/cli.js',
    '/opt/verdaccio/bin/verdaccio'
  ]
}
```

Since there is no `node_modules` anymore, how does yarn2 does the trick there 🤔 ?

Reply:
I think is because I'm not copying the yarn cache on this step
```
COPY --from=builder /opt/verdaccio-build .
```

Reply:
Here is a diff that fixes the docker image
```diff
diff --git a/Dockerfile b/Dockerfile
index 87294522..d570c7bf 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -15,6 +15,7 @@ COPY . .
 RUN yarn config set npmRegistryServer $VERDACCIO_BUILD_REGISTRY && \
     yarn config set enableProgressBars false && \
     yarn config set enableTelemetry false && \
+    yarn config set enableGlobalCache false && \
     yarn install && \
     yarn code:docker-build && \
     yarn cache clean && \
@@ -56,4 +57,4 @@ VOLUME /verdaccio/storage
 
 ENTRYPOINT ["uid_entrypoint"]
 
-CMD $VERDACCIO_APPDIR/bin/verdaccio --config /verdaccio/conf/config.yaml --listen $VERDACCIO_PROTOCOL://0.0.0.0:$VERDACCIO_PORT
+CMD node -r ./.pnp.js $VERDACCIO_APPDIR/bin/verdaccio --config /verdaccio/conf/config.yaml --listen $VERDACCIO_PROTOCOL://0.0.0.0:$VERDACCIO_PORT

```

Reply:
`node -r ./.pnp.js` <-- cool stuff :) I din't know this one. Thanks again. Ufff it was a long ride.